### ISSUE TITLE
UX - Cannot drag project folder into tree-view

### ISSUE THREAD

**Issue by mnquintana** [atom/tree-view#480]
_From @jsphstls on April 27, 2015 16:6_

**To reproduce:**
Open Atom and clear any project folders or open files.
Open Finder and drag then drop a folder into the Atom tree-view.

**Current result:**
Project folder will not be added to tree-view.

**Expected result:**
Project folder would be added to tree-view.

**Suggested change:**
Allow project folders to be dragged into tree-view.

It is odd that you cannot add a project folder this way since the tree-view is where project folders are navigated. Instead, you can drag a folder into an editor pane, which is where individual files are edited.

_Copied from original issue: atom/atom#6535_

**Comment by mnquintana** [comment]
_From @andrewrothman on May 4, 2015 7:23_

I also find this odd. Definitely tried it a few times with frustration before realizing it couldn't be done.

**Comment by moobee** [comment]
Would be nice to have :+1:

**Comment by Sogl** [comment]
+1

**Comment by Tyriar** [comment]
I have this working locally (without an `ondragover` hint), the specs will build on the event-helpers changes being added in https://github.com/atom/tree-view/pull/604

**Comment by joel** [comment]
Can we disable this feature? Thanks

**Comment by nahuelhds** [comment]
+1

**Comment by ThornWinters** [comment]
+1

**Comment by tofof** [comment]
+1

**Comment by billybonks** [comment]
@joel im not sure what you are talking about, this feature doesn't exist.

**Comment by joel** [comment]
@billybonks oh!!! I see, I though it was moving folders into the tree-view itself! And this bother me a lot because I do that by accident often, thank to git to save me.... But you're right isn't the feature we are talking about here, however I don't really see the benefit of this one, but it doesn't matter 😆

**Comment by xinmeng1** [comment]
+1 very useful feature. I previous use sublime which has this feature.

**Comment by 14nd90** [comment]
+1

**Comment by yramocan** [comment]
+1

**Comment by eddie-lopez** [comment]
+1

**Comment by procmail** [comment]
+1 any progress on this?

**Comment by shenkan** [comment]
this is definitely needed, as is the other issue on dragging a pane off the window to open the file new editor window

**Comment by thkdog** [comment]
+1

**Comment by rclations** [comment]
#1 biggest pain point of using Atom

**Comment by saresq** [comment]
It would be cool if they added this functionality, it would save so much time!

**Comment by RiccardoZaffalon** [comment]
+1 Hoping to see this feature soon!
In the mean time I drag folders in the editor pane (where the code is), which opens the project in a new Atom window. I then drag that folder from the new window's tree-view to the old tree-view.
Hope this helps!

**Comment by johnmarian** [comment]
+1

**Comment by getpavilion** [comment]
+1

**Comment by jsphstls** [comment]
This issue will be two years old later this month. Every time I see a +1 it brightens my day. UX is pretty important for me. I use Atom for the UI customization most of all. In all seriousness, this is not the worst issue, but I think that the following ones are. I may was well use this aging thread for some shout-outs:

1. If you open a single file on a network drive, Atom will attempt to "read" the rest of the directory regardless of size or how much data I have left that month. It completely freezes over my VPN. I have no way of turning off this feature. I have to go into "90's mode" where I copy a file from the network, make changes, and put it back. Is "zen coding" about slowing things down?

2. Crashing on startup. How did this app leave whatever "beta" with that? Everyone can easily blame the plugins. "Have you tried turning off your plugins?" No, I am trying to use the app as it was designed, my bad. I have looked for every setting that might get Atom to open "quietly." My favorite trick is opening it once, opening a new window, closing the first window with the helpful alternative to "wait", and then hoping no one noticed that. I have seen some more intensive solutions but remember that this is a UX thread.

I have found these issues documented in a few places so I'd rather not add to the pile. I want to promote this tool but it feels like I am signing others up for trouble that I should be smart enough to get out of on my own. If I actually paid for this I would use content like this when requesting a refund. But, I didn't, so I'll just vent.

**Comment by mourningking** [comment]
On Mac, you can File → Add Project Folder... and then drag and drop folder there. Hope it helps some of you guys. Still, would be nice to drop folder directly to the tree-view.

**Comment by sylrs** [comment]
+1

**Comment by burakunaltay** [comment]
+1

**Comment by ndrs92** [comment]
+1

**Comment by zhoukekestar** [comment]
Maybe this is what you need, a tiny script to implement it. 
Attention! This script is not fully tested.
Add this to your `File` -> `Init Script`. 
```coffee
fs = require('fs')
treeView = document.querySelector('ol.tree-view.full-menu.list-tree')
if treeView
  treeView.addEventListener 'drop', (event) ->
    if !event.dataTransfer.files or event.dataTransfer.files.length == 0
      return
    try
      path = event.dataTransfer.files[0].path
      stat = fs.statSync(path)
      if stat.isDirectory()
        atom.project.addPath path
    catch e
      console.error e
    return
  treeView.addEventListener 'dragover', (event) ->
    event.preventDefault()
    return
```
![Demo](https://cloud.githubusercontent.com/assets/7157346/25646208/a705fed0-2fe8-11e7-9e1d-941319e3a6f3.gif)

**Comment by missmatsuko** [comment]
Weirdly, dragging a folder into the tree view pane opens it as a project for me for the first project only. Additional projects will not be added to the tree view or opened at all. I'd prefer if it would add it to the tree view.

**Comment by kent10ou** [comment]
+1

**Comment by hoperyy** [comment]
+1

**Comment by otonielguajardo** [comment]
+1

**Comment by MiralDesai** [comment]
+1

**Comment by nalt** [comment]
+1

**Comment by tifa2UP** [comment]
+1

**Comment by Sixdd6** [comment]
+1

Just started using Atom after switching from SublimeText and really miss this feature

**Comment by bneigher** [comment]
+1

**Comment by 50Wliu** [comment]
With the implementation in #1209, you can now drag an external folder to an _empty_ section of the Tree View below the last project folder to create a new project folder.  This is similar to the GIF that @zhoukekestar provided.

**Comment by andreasmb** [comment]
+1

**Comment by aumhaa** [comment]
What about dragging out of tree view to finder?  I really miss this capability from Textmate :(

**Comment by Arcanemagus** [comment]
@aumhaa That is currently only released in the beta, are you on v1.31.0-beta2?
